Frontend/app/src/main/java/io/happybugs/happybugs/activity/MainActivity.java

```diff
@@ -62,56 +62,38 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
-        btnStartReport = (Button) findViewById(R.id.button_startreport);
-        btnStartReport.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                startActivity(new Intent(currContext, ReportActivity.class));
-            }
-        });
-
-        BottomNavigationView bottomNavigationView =
-                (BottomNavigationView) findViewById(R.id.bottom_navigation);
-        bottomNavigationView.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener);
-
-        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
-        setSupportActionBar(toolbar);
-
-        DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
-        ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
-                this, drawer, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close);
-        drawer.addDrawerListener(toggle);
-        toggle.syncState();
-
-        NavigationView navigationView = (NavigationView) findViewById(R.id.nav_view);
-        navigationView.setNavigationItemSelectedListener(this);
-
         Retrofit rfInstance = RetrofitInstance.getInstance(currContext);
         APIInterface service = rfInstance.create(APIInterface.class);
+        requestReportList = service.getReportList();
+
+        navigationSettings();
+        showReportList();
+    }
+
+    public void showReportList() {
+        final ListView reportListView = (ListView) findViewById(R.id.report_listview);
+        final ReportListViewAdapter reportListViewAdapter = new ReportListViewAdapter();
 
-        Call<UserReportList> requestReportList = service.getReportList();
         requestReportList.enqueue(new Callback<UserReportList>() {
             @Override
             public void onResponse(Call<UserReportList> call, Response<UserReportList> response) {
-                List<UserReportList> userReportLists = response.body().getData();
-                finish();
+                List<UserReportItem> userReportList = response.body().getData();
+
+                if (!(userReportList.isEmpty())) {
+                    invisibleHomeIntroContents();
+
+                    for (UserReportItem item : userReportList) {
+                        reportListViewAdapter.addItem(item.getWhat());
```

Comment:
maybe expose addAll method in adapter class and use it? Not sure how much efficient this will be because we'll only have small number of reports, but I think this will be a good exercise.